lib/onigumo/parser.ex

```diff
@@ -0,0 +1,26 @@
+defmodule Onigumo.Parser do
+  @moduledoc """
+  Web scraper
+  """
+  @behaviour Onigumo.Component
+
+  @impl Onigumo.Component
+  def main(root_path) do
+    root_path
+    |> list_downloaded()
+    |> Enum.map(&IO.puts(&1))
+
+    :ok
+  end
+
+  defp list_downloaded(path) do
+    path
+    |> File.ls!()
+    |> Enum.filter(&is_downloaded(&1))
+  end
+
+  defp is_downloaded(path) do
+    suffix = Application.get_env(:onigumo, :downloaded_suffix)
```

Comment:
The `donwloaded_suffix` should be constant during running of code, may the `compile_env` will be more suitable to ensure to use the value which is really specified in configuration and do not be aware that the value was somewhere changed during running of code.

Reply:
`:downloaded_suffix` is a (user-)configurable value. If it were compiled, it wouldn’t be possible to change it by editting the configuration file without recompilation. [get_env](https://hexdocs.pm/elixir/Application.html#get_env/3) makes sense here.

It’s not like [compile_env](https://hexdocs.pm/elixir/Application.html#get_env/3) in the module [mapping](https://github.com/Glutexo/onigumo/blob/0d890b963ba18cca8d819bff58bd691adf27b458/lib/cli.ex#L3) in the CLI module. It’s “configurable” only so it can be mocked in the test.